src/main/java/org/opentest4j/AssertionFailedError.java

```diff
@@ -34,72 +32,50 @@
 
 	private static final long serialVersionUID = 1L;
 
-	private final Serializable expected;
-	private final Serializable actual;
+	private final ValueWrapper expected;
+	private final ValueWrapper actual;
 
 	public AssertionFailedError() {
 		this(null);
 	}
 
 	public AssertionFailedError(String message) {
-		this(message, Undefined.INSTANCE, Undefined.INSTANCE);
+		this(message, null);
 	}
 
-	public AssertionFailedError(String message, Serializable expected, Serializable actual) {
+	public AssertionFailedError(String message, Object expected, Object actual) {
 		this(message, expected, actual, null);
 	}
 
 	public AssertionFailedError(String message, Throwable cause) {
-		this(message, Undefined.INSTANCE, Undefined.INSTANCE, cause);
+		this(message, (ValueWrapper) null, (ValueWrapper) null, cause);
```

Comment:
You're right these casts are redundant because the `ValueWrapper` constructor is more specific than the `Object` constructor (cf. [JLS §15.12.2.5](http://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.12.2.5)). I'll remove them and add a commit.